pkg/opentelemetry-mapping-go/otlp/metrics/metrics_translator.go

```diff
@@ -141,15 +139,99 @@ func NewTranslator(set component.TelemetrySettings, attributesTranslator *attrib
 	}
 
 	cache := newTTLCache(cfg.sweepInterval, cfg.deltaTTL)
+	logger := set.Logger.With(zap.String("component", "metrics translator"))
 
-	return &Translator{
+	t := &defaultTranslator{
 		prevPts:              cache,
-		logger:               set.Logger.With(zap.String("component", "metrics translator")),
+		logger:               logger,
 		attributesTranslator: attributesTranslator,
 		cfg:                  cfg,
+	}
+	// Use custom mapper if provided, otherwise create a DefaultMapper
+	if cfg.customMapper != nil {
+		t.mapper = cfg.customMapper
+	} else {
+		t.mapper = newDefaultMapper(cache, logger, cfg)
+	}
+	return t, nil
```

Comment:
nit: can avoid `else` here
```suggestion
		return t, nil
	}
	t.mapper = newDefaultMapper(cache, logger, cfg)
	return t, nil
```